bundles/org.openhab.binding.exec/src/main/java/org/openhab/binding/exec/internal/handler/ExecHandler.java

```diff
@@ -298,4 +350,76 @@ public void run() {
         return new String[] { type, pattern };
     }
 
-}
+    /**
+     * Transforms the command string into an array.
+     * Either invokes the shell and passes using the "c" option
+     * or (if command already starts with one of the shells) splits by space.
+     *
+     * @param shell (path), picks to first one to execute the command
+     * @param "c"-option string
+     * @param command to execute
+     * @return command array
+     */
+    protected String[] createCmdArray(String[] shell, String cOption, String commandLine) {
+        boolean startsWithShell = false;
+        for (String sh : shell) {
+            if (commandLine.startsWith(sh+" ")) {
+                startsWithShell = true;
+                break;
+            }
+        }
+
+        if (!startsWithShell) {
+            return new String[]{shell[0], cOption, commandLine};
+        } else {
+            logger.debug("Splitting by spaces");
+            try {
+                return commandLine.split(" ");
+            } catch (PatternSyntaxException e) {
+                logger.warn("An exception occurred while splitting '{}' : '{}'",
+                        commandLine, e.getMessage());
+                updateState(RUN, OnOffType.OFF);
+                updateState(OUTPUT, new StringType(e.getMessage()));
+                return new String[] {};
+            }
+        }
+    }
+
+
+
+    /**
+     * Contains information about which operating system openHAB is running on.
+     * Found on https://stackoverflow.com/a/31547504/7508309, slightly modified
+     *
+     * @author Constantin Piber (for Memin) - Initial contribution
+     */
+    public enum OS {
+        WINDOWS, LINUX, MAC, SOLARIS, UNKNOWN, NOT_SET
+    };
+
+    private static OS os = OS.NOT_SET;
+
+    public static OS getOperatingSystemType() {
+        if (os == OS.NOT_SET) {
+            String operSys = System.getProperty("os.name").toLowerCase();
+            if (operSys.contains("win")) {
+                os = OS.WINDOWS;
+            } else if (operSys.contains("nix") || operSys.contains("nux")
+                    || operSys.contains("aix")) {
+                os = OS.LINUX;
+            } else if (operSys.contains("mac")) {
+                os = OS.MAC;
+            } else if (operSys.contains("sunos")) {
+                os = OS.SOLARIS;
+            } else {
+                os = OS.UNKNOWN;
+            }
+        }
+        return os;
+    }
+
+    public static String getOperatingSystemName() {
+        return System.getProperty("os.name");
+    }
+
+}
```

Comment:
Please run `mvn spotless:apply` after the code issues are done.

Reply:
What exactly does that do?

Reply:
Okay so it runs some more tests, I get
[ERROR] Failed to execute goal com.diffplug.spotless:spotless-maven-plugin:1.24.3:apply (default-cli) on project org.openhab.binding.feed.tests: Execution default-cli of goal com.diffplug.spotless:spotless-maven-plugin:1.24.3:apply failed: Overlapping text edits -> [Help 1]
What do I do? Can't find any info on that

Reply:
@J-N-K

Reply:
Reset the changes in other bindings and use `mvn spotless:apply -pl bundles\org.openhab.binding.exec` to only re-format the exec binding.